### PR TITLE
[Added Fix ] Enable parallel execution of OVC Python API python tests 

### LINKED ISSUE #20922: [Good First Issue]: Enable parallel execution of OVC Python API python tests
### Context

Current CI (https://github.com/openvinotoolkit/openvino/blob/c0381ab08d2fcdc51c621196e935aa02c64a4b50/.github/workflows/linux.yml#L869) runs all the python tests sequentially one by one and it takes too much time.
The idea is to run all those tests in parallel by using pytest-xdist plugin, but tests should be modified for that.
Please find bellow the common tests issues:
- Random inputs generation as parametrization arguments. Solution: all random test arguments should be generated inside of the tests 
- Tests create, use and remove the same artifacts (temporary data, models, etc). Hot fix solution: all test data should be generated in the system temporary directory (TMPDIR) which can be unique for each thread. Long term solution: all test data should be generated in the unique temporary directory or with unique names.

### What needs to be done?

- Modify OVC Python API tests to solve issues described in the Context. Tests are located here - https://github.com/openvinotoolkit/openvino/tree/master/tests/layer_tests/ovc_python_api_tests
- Enable parallel execution of those test in the CI (https://github.com/openvinotoolkit/openvino/blob/4a0098b26ad6c4cbae97fbe34505fa63c08036eb/.github/workflows/linux.yml#L1001C15-L1001C35)

### Example Pull Requests

https://github.com/openvinotoolkit/openvino/pull/20613

### Resources

- [Contribution guide - start here!](https://github.com/openvinotoolkit/openvino/blob/master/CONTRIBUTING.md)
- [Intel DevHub Discord channel](https://discord.gg/wPuqAujS) - engage in discussions, ask questions and talk to OpenVINO developers
- [What is OpenVINO?](https://github.com/openvinotoolkit/openvino#what-is-openvino-toolkit)
- [Blog post on contributing to OpenVINO](https://github.com/openvinotoolkit/openvino/blob/master/CONTRIBUTING.md)
- [User documentation](https://docs.openvino.ai/)

### Contact points

@mryzhov @akashchi

### Ticket

_No response_

### PR DESCRIPTION
**Pull request by mahajanparth**
### Details:
 - Enable parallel execution of OVC Python API python tests
 - The issue can be found on the [#20922](https://github.com/openvinotoolkit/openvino/issues/20922)
 

### Tickets:
 -  Closes #20922

### CODE REVIEW

**Review by rkazants — COMMENTED** [review]
please add `-n logical` here for OVC tests: https://github.com/openvinotoolkit/openvino/blob/4a0098b26ad6c4cbae97fbe34505fa63c08036eb/.github/workflows/linux.yml#L1001C15-L1001C35

It will activate parallel execution

**Comment by mahajanparth** [comment]
@rkazants  i might be asking a very silly question but the files are very different for the master branch for  .github/workflow/linux.yml and the checkout version of the commit that you are pointing for the same file. Do i meld the code for that particular detached head and push the code again instead of the master branch?

**Comment by mahajanparth** [comment]
@rkazants  i might be asking a very silly question but the files are very different for the master branch for  .github/workflow/linux.yml and the checkout version of the commit that you are pointing for the same file. Do i meld the code for that particular detached head and push the code again instead of the master branch?

**Comment by rkazants** [comment]
> @rkazants i might be asking a very silly question but the file linux.yml is very different for the master branch for .github/workflow/linux.yml and the detached head version of the commit that you are pointing for the same file linux.yml. Do i meld the code for that particular detached head again and push the code instead of the master branch?

Hi @mahajanparth, indeed this code was moved some time ago. Find it here: https://github.com/openvinotoolkit/openvino/blob/master/.github/workflows/job_python_unit_tests.yml#L171

Best regards,
Roman

**Comment by mahajanparth** [comment]
@rkazants  thanks for the quick help. It's done !!!

**Comment by rkazants** [comment]
build_jenkins

**Comment by mahajanparth** [comment]
@rkazants I can see the ovc_python_api test cases are being executed and passed successfully. Is there anything required from my end to close this ticket?

**Comment by mryzhov** [comment]
Could you please also enable parallel execution for windows pipeline?

https://github.com/openvinotoolkit/openvino/blob/e8b30887cda05efd2cc6c9037c124753f9885e50/.github/workflows/windows.yml#L691

File: tests/layer_tests/ovc_python_api_tests/test_complex_params.py [hunk 1]
```diff
@@ -78,7 +78,7 @@ def create_tf_model_single_input_output(tmp_dir):
         {'params_test': {'input': [PartialShape([2, 3, 4]), [2, 3, 4], [Dimension(2), Dimension(3), Dimension(4)]]},
          'params_ref': {'input_shape': "[2,3,4],[2,3,4],[2,3,4]", 'input': 'Input1:0,Input2:0,Input3:0'}},
         {'params_test': {'input': [PartialShape([1, 3, -1, -1]), [1, 3, -1, -1]]},
-         'params_ref': {'input_shape': "[1,3,?,?],[1,3,?,?]", 'input': 'Input1:0,Input2:0'}},
```
Comment (mryzhov) [inline tests/layer_tests/ovc_python_api_tests/test_complex_params.py:81]: looks like unnecessary сhange, could you please remove?

Comment (mahajanparth) [reply tests/layer_tests/ovc_python_api_tests/test_complex_params.py]: Done @mryzhov

**Comment by mahajanparth** [comment]
> Could you please also enable parallel execution for windows pipeline?
> 
> https://github.com/openvinotoolkit/openvino/blob/e8b30887cda05efd2cc6c9037c124753f9885e50/.github/workflows/windows.yml#L691

Also Done

**Review by mryzhov — APPROVED** [review]
LGFM, @rkazants please take a look at the tests modifications

File: tests/layer_tests/ovc_python_api_tests/test_tf_output_order.py [hunk 2]
```diff
@@ -67,13 +67,15 @@ class TestTFInputOutputOrder():
...
-
-    @pytest.mark.parametrize("save_to_file, create_model_method, compare_model_method", [
+    test_data=[
         (False, create_net_list, check_outputs_by_order),
         (False, create_net_dict, check_outputs_by_names),
         pytest.param(True, create_net_list, check_outputs_by_order, marks=pytest.mark.xfail(reason='124436')),
         pytest.param(True, create_net_dict, check_outputs_by_names, marks=pytest.mark.xfail(reason='124436')),
-    ])
+    ]
+    test_ids = ["Test{}".format(id) for id in range(len(test_data))]
```
Comment (rkazants) [inline tests/layer_tests/ovc_python_api_tests/test_tf_output_order.py:76]: why do we need `test_ids`? I think it was used for debugging and now can be removed. Right, @mahajanparth?

**Comment by mlukasze** [comment]
hey @mahajanparth will you find a time to make these small corrections Roman mentioned above?
We are really close to merge it and would be cool if you would find some time to polish PR to that point :)

**Pull request closed without merging**